cli/main.go

```diff
@@ -1264,34 +1263,121 @@ func ValidateSeedFile(schemaFile string, seedFileName string, schemaType constan
+	fmt.Fprintf(os.Stderr, "INFO: Checking for reserved variable namem collisions...\n")
+	seed := SeedFromManifestFile(seedFileName)
+
+	// Grab all sclar resource names (verify none are set to OUTPUT_DIR)
+	var allocated []string
+	if seed.Job.Interface.Resources.Scalar != nil {
+		for _, s := range seed.Job.Interface.Resources.Scalar {
+			name := GetNormalizedVariable(s.Name)
+			allocated = append(allocated, "ALLOCATED_"+strings.ToUpper(name))
+			if IsReserved(s.Name, nil) {
+				buffer.WriteString("ERROR: Scalar Resource Name " + s.Name +
+					" is a reserved variable.\n")
+			}
+		}
+	}
+
+	if seed.Job.Interface.InputData.Files != nil {
+		for _, f := range seed.Job.Interface.InputData.Files {
+			// check against the ALLOCATED_* and OUTPUT_DIR
+			if IsReserved(f.Name, allocated) {
+				buffer.WriteString("ERROR: InputData File Name " + f.Name +
+					" is a reserved variable.\n")
+			}
+		}
+	}
+
+	if seed.Job.Interface.InputData.Json != nil {
+		for _, f := range seed.Job.Interface.InputData.Json {
+			if IsReserved(f.Name, allocated) {
+				buffer.WriteString("ERROR: InputData JSON Name " + f.Name +
+					" is a reserved variable.\n")
+			}
+		}
+	}
+
+	if seed.Job.Interface.OutputData.Files != nil {
+		for _, f := range seed.Job.Interface.OutputData.Files {
+			// check against the ALLOCATED_* and OUTPUT_DIR
+			if IsReserved(f.Name, allocated) {
+				buffer.WriteString("ERROR: OutputData File Name " + f.Name +
+					" is a reserved variable.\n")
+			}
+		}
+	}
+
+	if seed.Job.Interface.OutputData.JSON != nil {
+		for _, f := range seed.Job.Interface.OutputData.JSON {
+			// check against the ALLOCATED_* and OUTPUT_DIR
+			if IsReserved(f.Name, allocated) {
+				buffer.WriteString("ERROR: OutputData JSON Name " + f.Name +
+					" is a reserved variable.\n")
+			}
+		}
+	}
+
+	if seed.Job.Interface.Mounts != nil {
+		for _, m := range seed.Job.Interface.Mounts {
+			// check against the ALLOCATED_* and OUTPUT_DIR
+			if IsReserved(m.Name, allocated) {
+				buffer.WriteString("ERROR: Mounts Name " + m.Name +
+					" is a reserved variable.\n")
+			}
+		}
+	}
+
+	if seed.Job.Interface.Settings != nil {
+		for _, s := range seed.Job.Interface.Settings {
+			// check against the ALLOCATED_* and OUTPUT_DIR
+			if IsReserved(s.Name, allocated) {
+				buffer.WriteString("ERROR: Settings Name " + s.Name +
+					" is a reserved variable.\n")
+			}
+		}
+	}
+
+	// Return error if issues found
+	if buffer.String() != "" {
+		return errors.New(buffer.String())
+	}
 
 	// Validation succeeded
-	fmt.Fprintf(os.Stderr, "SUCCESS: %s is valid.\n\n", seedFileName)
+	fmt.Fprintf(os.Stderr, "SUCCESS: No errors found. %s is valid.\n\n", seedFileName)
 	return nil
 }
 
+//IsReserved checks if the given string is one of the reserved variable names
+func IsReserved(name string, allocated []string) bool {
+	reserved := name == "OUTPUT_DIR"
+
+	if allocated != nil {
+		for _, s := range allocated {
+			if s == strings.ToUpper(name) {
```

Comment:
Instead of just passing string through ToUpper, they should all be passed through `GetNormalizedVariable`. That function has the logic to appropriately sanitize any allowed variable and is used when injecting the settings... we need to extend its use to inject variables for all members being deconflicted here.